python-job-crawler.py

```diff
@@ -1,88 +1,143 @@
-    # chromeDriverPath = '/Users/sueellenmisky/.wdm/drivers/chromedriver/mac64/127.0.6533.119/chromedriver-mac-x64/chromedriver'
-    # Ensure chromeDriverPath.py is in your .gitignore
+    options_list = ['--headless', '--no-sandbox', '--disable-dev-shm-usage']
+    for opt in options_list:
+        chrome_options.add_argument(opt)
+
+    # Initialize WebDriver using ChromeDriverManager
     service = Service(chromeDriverPath)
     driver = webdriver.Chrome(service=service, options=chrome_options)
 
-    keywords_found_count = 0
+    # Get the number of match
+    kws_found_count = 0
     for careerPage in careerPages:
-        if crawl_careerPage(careerPage, driver):
-            keywords_found_count += 1
-        # Add a delay between requests
-        time.sleep(1)  
+        if crawl_careerPage(careerPage, driver, keywords):
+            kws_found_count += 1
+        time.sleep(1)
 
     driver.quit()
-    print(f"Keywords were found in job listings at {keywords_found_count} out of {len(careerPages)} career pages.")
-    print(f"💤 Web crawler has finished.")
+    logging.info(f"Keywords were found in job listings at {kws_found_count} out of {len(careerPages)} career pages.")
 
-def close_dialogs(driver):
+
+def close_dialogs(driver: webdriver.Chrome) -> None:
+    """
+    Tries to close any potential pop-up dialogs that may appear on the page.
+
+    Args:
+        driver (webdriver.Chrome): The Selenium WebDriver instance used for browsing.
+
+    Returns:
+        None
+    """
     try:
-        # If there's a dialog box with a close button with class .close-button or .close
-        # Add more class names as you encounter them
-        close_button = driver.find_element(By.CSS_SELECTOR, '.close-button' | '.close')
+        # Modify the CSS selector to find common close buttons, adjust as needed
+        close_button = WebDriverWait(driver, PAGE_LOADING_TIME_SLEEPER).until(
+            EC.element_to_be_clickable((By.CSS_SELECTOR, '.close-button, .close'))
+        )
         if close_button:
             close_button.click()
+            logging.info("Closed a dialog on the page.")
+    except TimeoutException:
+        logging.debug("No dialog found to close.")
     except Exception as e:
-        # No dialog to close or unable to find the close button
-        pass
+        logging.error(f"An unexpected error occurred: {e}")
 
-def crawl_careerPage(url, driver):
-    driver.get(url)
-    time.sleep(5)  # Wait for initial load
 
-    # Handle any potential dialogs
-    close_dialogs(driver)
+def crawl_careerPage(url: str, driver: webdriver.Chrome, kws: List[str]) -> bool:
+    """
+    Crawls a given career page URL to search for keywords in the job listings.
+
+    Args:
+        url (str): URL of the career page to visit.
+        driver (webdriver.Chrome): Selenium WebDriver instance used for browsing.
+        kws (List[str]): the kws to look for on page at URL
```

Comment:
Again, replace the `kws` to `keywords`.
keywords (List[str]): the keywords to look for on page at URL